### ISSUE TITLE
Inaccuracy regarding cost of virtual method calls

### ISSUE THREAD

**Issue by qtormanen** [munificent/game-programming-patterns#124]
In the Data Locality section of the Optimization chapter, there is a sidebar next to the Keep in Mind sub-section that says the following:

"The other half of interfaces is virtual method calls. Those require the CPU to look up an object’s vtable, and then find the pointer to the actual method to call there. So, again, you’re chasing a pointer and often causing a cache miss, though this time it’s an instruction cache miss, since you’re reading code from RAM."

In C++ at least, the extra cost of making the virtual method call is going to be in loading two pointers--the vtable and then the function pointer--and both of these are _data_ loads and can only cause _data_ cache misses, not _instruction_ cache misses. The instruction pointer fetch will jump directly from the code that's loading the vtable and function pointers to the code implementing the function, which is no different than a non-virtual method call. Side note: there is more cost in terms of the instruction pipeline having to stall until it knows where the code is jumping, but I think it's inaccurate to suggest that the extra cache miss referenced in your sidebar will be on the instruction cache.

Bottom line, I think the phrase in the side bar "though this time it’s an instruction cache miss, since you’re reading code from RAM" is inaccurate and should be changed or removed.
